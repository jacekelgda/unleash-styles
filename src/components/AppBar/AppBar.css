.AppBar {
	display: flex;
	justify-content: space-between;
	align-items: center;
	border-bottom: 1px solid #e8e8e8;
	padding: 0 18px;
	height: 70px;

	position: absolute;
	z-index: 1000;
	top: 0;
	width: calc(100% - 36px); /* 18px left/right padding */
	background-color: #fff;
}

@media screen and (min-width: 768px) {
	.AppBar {
		padding: 0 28px;
		width: calc(100% - 56px); /* 28px left/right padding */
	}
}
@media screen and (min-width: 1024px) {
	.AppBar {
		padding: 0 70px;
		width: calc(100% - 140px); /* 70px left/right padding */
	}
}
