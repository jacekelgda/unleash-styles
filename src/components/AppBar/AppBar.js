import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './AppBar.css'

class AppBar extends Component {
  render() {
    const { appBarLogo } = this.props
    return (
      <header className={styles.AppBar}>
        {appBarLogo}
        {this.props.children}
    	</header>
    )
  }
}

AppBar.propTypes = {
  appBarLogo: PropTypes.object
}

export default AppBar
