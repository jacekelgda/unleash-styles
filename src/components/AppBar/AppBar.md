App bar with logo:

    <AppBar appBarLogo={<Logo />} />

App bar with logo, menu:

    <AppBar appBarLogo={<Logo />}>
      <Tabs>
        <Tab label="home" />
        <Tab label="my path" />
        <Tab label="profile" />
        <Tab label="goals" />
        <Tab label="skills" />
      </Tabs>
      <Menu menuButton={<Avatar type="small" label="NO" />}>
        <MenuItem label="nikita" type="important" />
        <MenuItem label="nikita.ormonov@eleken.co" type="email" />
        <MenuItem label="edit profile" type="link" />
        <MenuItem label="sign out" type="link" />
      </Menu>
    </AppBar>
