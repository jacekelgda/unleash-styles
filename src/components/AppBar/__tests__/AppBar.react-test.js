import React from 'react'
import renderer from 'react-test-renderer'

import AppBar from '../AppBar'

test('AppBar renders correctly', () => {
  const tree = renderer.create(
    <AppBar/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
