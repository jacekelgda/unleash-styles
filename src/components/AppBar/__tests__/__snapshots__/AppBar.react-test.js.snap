// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
<header
  className="AppBar"
/>
`;
